src/app/api/routes/devices.py

```diff
@@ -56,6 +57,13 @@ async def update_device(payload: DeviceIn, device_id: int = Path(..., gt=0)):
     """
     Based on a device_id, updates information about the specified device
     """
+    if payload.login is not None:
+        updated_device = await crud.fetch_one(devices, {"id": device_id})
+        if updated_device is not None and payload.login != updated_device["login"]:
+            await check_for_access_login_existence(payload.login)
+            updated_acccess = await crud.fetch_one(accesses, {"login": updated_device["login"]})
```

Comment:
over here, if `check_for_access_login_existence` returns the entry, we can reduce overhead

Reply:
Unless I have overlooked, we don't want to alter the access object found by `check_for_access_login_existence` so there is no use in retrieving it. As a matter of fact, *IF* there is a found entry, then we should stop the function and raise an error.

Reply:
Or are you suggesting to put the retrieval of the real updated_access into the ̀check_for_access_login_existencèfunction?

Reply:
nevermind, I overlooked that we are fetching another ID! Sorry about that!